src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java

```diff
@@ -405,13 +406,13 @@ private boolean checkCurrentResourcesStatus(
   }
 
   public synchronized boolean lock(
-    Set<LockableResource> resources, Run<?, ?> build, @Nullable StepContext context) {
+    List<LockableResource> resources, Run<?, ?> build, @Nullable StepContext context) {
```

Comment:
so I'm by no means a java person, but I am a dev who uses Jenkins.
can someone explain to me why this param type change from `Set` to `List` isn't considered a breaking change, or even worth a callout in the release notes?

it breaks groovy jobs like mine that call `lock()` directly because it was previously being created as a `Set` to match the API.

is this a release notes/documentation failure, or am I misusing the plugin maybe?

Reply:
Hi @joshsleeper 

Definitely a failure to report in the release notes. While I am an experienced Java developer, I only just recently started contributing to Jenkins plugins. My experience with them is to only use operations exposed as steps, and I initially did not realize some people were calling internal plug-in methods directly and had backward compatibility expectations on them. I'll definitely be more careful to flag these breaking changes going forward and for now I can only apologize.

Gaspard

Reply:
no anger here and plenty of understanding~
this isn't the first upgrade of this plugin that's busted things for us, so figured we were better off calling it out in the hopes that it doesn't happen every time!

I would suggest, if possible, to consider updating the release notes to mention the breaking change~

that aside, jenkins plugin maintenance def isn't a job folks do for glory so thanks for all your hard work!